micromamba.el

```diff
@@ -157,6 +195,51 @@ Returns an alist with the following keys:
       (vars-export . ,vars-export)
       (scripts . ,scripts))))
 
+(defun micromamba--env-dir-is-valid (candidate)
+  "Confirm that CANDIDATE is a valid conda environment."
+  (let ((dir (file-name-as-directory candidate)))
+    (and (not (s-blank? candidate))
+         (f-directory? dir)
+         (or (f-directory? (concat dir micromamba-env-executables-dir))
+             (f-directory? (concat dir micromamba-env-meta-dir))))))
+
+(defun micromamba--contains-env-yml? (candidate) ;; adapted from conda.el
+  "Does CANDIDATE contain an environment.yml?"
+  (f-exists? (f-expand "environment.yml" candidate)))
+
+(defun micromamba--find-env-yml (dir) ;; adapted from conda.el
+  "Find an environment.yml in DIR or its parent directories."
+  ;; TODO: implement an optimized finder with e.g. projectile? Or a series of
+  ;; finder functions, that stop at the project root when traversing
+  (let ((containing-path (f-traverse-upwards 'micromamba--contains-env-yml? dir)))
+    (if containing-path
+        (f-expand "environment.yml" containing-path)
+      nil)))
+
+(defun micromamba--get-name-from-env-yml (filename) ;; adapted from conda.el
+  "Pull the `name` property out of the YAML file at FILENAME."
+  ;; TODO: find a better way than slurping it in and using a regex...
```

Comment:
Uhm... TODO? :slightly_smiling_face:

Reply:
I'd at least cache the results of this against `file-has-changed-p`. I'd imagine this would be pretty slow on HDD.